scripts/add-icon-data.js

```diff
@@ -88,85 +89,103 @@ const getIconDataFromAnswers = (answers) => ({
 
 const answers = {};
 
-answers.title = await input({
-  message: 'Title',
-  validate: titleValidator,
-});
-
-answers.hex = await input({
-  message: 'Hex',
-  validate: hexValidator,
-  transformer: hexTransformer,
-});
-
-answers.source = await input({
-  message: 'Source',
-  validate: sourceValidator,
-});
+try {
```

Comment:
Please, don't use a `try` / `catch` block. That would silently bypass all other possible exceptions. Use `process.on('SIGINT', () => {});` instead, see this thread on StackOverflow: https://stackoverflow.com/a/71543784/9167585

In this case, would you mind to add other signal interruptions? Adding SIGINT, SIGQUIT and SIGKILL should be fine.

Reply:
I think @inquirer/prompts intercepts the SIGINT and all that? How do you feel about:

```js
} catch (error) {
  if (
    error.message.includes('force closed the prompt') ||
    error.message.includes('prompt denied')
  ) {
    console.log(chalk.red('\nAborted.'));
    process.exit(1);
  } else {
    throw error;
  }
}
```

Reply:
Where is the catch of `SIGINT` in inquirer? Could you link it? I don't see that in the link that you've shared.

Reply:
I think that inquirer has a really weird behaviour and that's the reason why you're throwing an exception here. If I press `Ctrl^C` in the `title` prompt, it raises an exception:

```
file:///home/mondeja/files/code/simple-icons/node_modules/@inquirer/input/node_modules/@inquirer/core/dist/esm/lib/create-prompt.mjs:36
                    reject(new Error(`User force closed the prompt with ${code} ${signal}`));
                           ^

Error: User force closed the prompt with 0 null
    at file:///.../simple-icons/node_modules/@inquirer/input/node_modules/@inquirer/core/dist/esm/lib/create-prompt.mjs:36:28
    at Emitter.emit (file:///.../simple-icons/node_modules/signal-exit/dist/mjs/index.js:67:19)
    at #processEmit (file:///.../simple-icons/node_modules/signal-exit/dist/mjs/index.js:236:27)
    at #process.emit (file:///.../simple-icons/node_modules/signal-exit/dist/mjs/index.js:187:37)
    at process.callbackTrampoline (node:internal/async_hooks:130:17)
```

This looks a really bad idea. What is the recommended solution given by inquirer maintainers? It is documented anywhere?

Reply:
Yep. That's why that's the best solution I could come up with. It could be worth it to switch to https://github.com/terkelg/prompts, which supports better cancel handling? They have a great [autocomplete prompt built-in](https://github.com/terkelg/prompts#autocompletemessage-choices-initial-suggest-limit-style). I'll make a new PR if that sounds good?

Reply:
I think that inquirer.js seems to be better maintained and to rewrite the whole script to make a change doesn't look worth to me. Just go with the licenses selection improvement in this, we can open an issue on inquirer.js to proper handle this case.